### ISSUE TITLE
Show domains requesting access below allowed domains for CORS

### ISSUE THREAD

**Issue by noveens** [owncloud/core#28877]
[Enhancement] [Discussion]

Visit Settings -> Personal -> Security

We can see the allowed Domains for CORS and add/remove domains.
But to make easy for developers that which all domains have been requesting access, Let's introduce another table below White-listed domains having 3 collumns `Domain, Timestamp, Allow Button` about the domains which have been requesting access.

Thoughts?
Any UI suggestions are welcome :-)

@PVince81 @Peter-Prochaska

**Comment by noveens** [comment]
To discuss:

- [ ] How many domains should we display?
- [ ] How many domains should we keep in DB?
- [ ] DB structure?

**Comment by PVince81** [comment]
You might also want a column telling what user it is.

**Comment by PVince81** [comment]
I assume that what you want is simply for an admin to get an overview about who requested what and the ability to manage/cancel such domains.

If yes, you already have the DB structure in the app config. You might need some custom queries to be able to retrieve the values more easily.

**Comment by PVince81** [comment]
Forget what I said, I misunderstood. For an admin overview of all user's whitelists, should be a separate ticket.

So here the use case is whenever a user has attempted to connect with CORS from an external app and the domain was not whitelisted yet, it would appear in the domain list of the user with a "pending approval" indicator. The user can then decide to either approve the domain or reject it.

This is a good idea.

Once this is in place, the next step is to provide a way for apps (like js-owncloud-client) to redirect to a settings page / popup inside OC in which the user can directly allow this domain. (a bit like the OAuth2) handshake.

And if OAuth is used, automatically whitelist whatever domain the originating app comes from.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

**Comment by cdamken** [comment]
I think some documentation would be good too

@settermjd  :smiley_cat:
